### ISSUE TITLE
DatePicker not working in 2.2 Android Version

### ISSUE THREAD

**Issue by patelakshay13890** [rey5137/material#12]
Date Picker not working in 2.2 Android Version only..
but my app do have compatibility to 2.2 so what's the solution for the same ?

**Comment by rey5137** [comment]
DatePicker has a known bug that I fixed in next version. And Material Library is intended to support Android 2.3+ . I think you should set the compatibility to 2.3 as there isn't many devices with Android 2.2.

**Comment by patelakshay13890** [comment]
Okay & thanx for quick response..!!

**Comment by patelakshay13890** [comment]
How can I use Button with RippleEffect without having delay in Click bcoz
in my app on Button click method should be triggered asap..
So can you help me out in this situation??

**Comment by rey5137** [comment]
You can set the `delayClick` attribute of Button to `false` . Please view [this page](https://github.com/rey5137/Material/wiki/Button) for details.

**Comment by patelakshay13890** [comment]
ok and how can I change BackgroundColor of Button ??

**Comment by patelakshay13890** [comment]
In XML is is showing Button with Colored Background but when I run the app
It doesnot show any Background Color ....& I have to use White Color as
Font Color

**Comment by rey5137** [comment]
Did you use the `android:background` attribute?
You should use `rd_background` attribute instead because Button with create a RippleDrawable object and set it as Button's background, override the value of `android:background` attribute. 
`rd_background` attribute accept color or drawable resourceId value.

**Comment by patelakshay13890** [comment]
I cannot achieve this as your solution..!!!
Can you provide sample for the same to change Button BackGround Color and
Its FontColor both in example ??

**Comment by rey5137** [comment]
Can you provide me the current xml and Ripple Effect style you use?

**Comment by patelakshay13890** [comment]
Can you provide one sample please for the same ??

**Comment by patelakshay13890** [comment]
How can I use Dialog in ActionBarActivity not int Fragment ??
How can I show Dialogue in ActionBarActivity bcoz there is not any method
like builder.show() or builder.dismiss(), so how can I do like this ??

**Comment by patelakshay13890** [comment]
How can I show Icon in the Dialogue box ???

**Comment by rey5137** [comment]
About sample, the demo app already has one in Button tab page. You can look at those raised Button in `fragment_button.xml` . Here is one of those Ripple style:

```xml
   <style name="RaiseButtonRippleStyle" parent="Material.Drawable.Ripple.Touch.Light">
        <item name="rd_background">@drawable/bg_bt_raise</item>
        ...
    </style>
``` 

You can see that I set `rd_background` attribute to a 9-patch drawable, in your case it should be a color value. I want to see your current code to know how it didn't work for you.

About using Dialog in Activity, did you mean that you want to show Dialog directly without putting it into a DialogFragment. If so, you just need to create a Dialog object, apply style and call `show()` method.

```java

    Dialog mDialog = new Dialog(context, styleId);

   //apply some style 
   mDialog.applyStyle(styleId)
        .title("Dialog title")
        .positiveAction("OK")
        .negativeAction("CANCEL");

    //show dialog
    mDialog.show();
```

Dialog.Buidler class only need when you want to store styling data and pass it to DialogFragment.

**Comment by patelakshay13890** [comment]
Here I have to Create Style For each button and I have 15 Button in One
Activity itself ..!!
So how can I do it with Simple Method ..??
BCoz I have to right 15 to 30 time different Style Method like

 <style name="RaiseButtonRippleStyle"
parent="Material.Drawable.Ripple.Touch.Light">
        <item name="rd_background">@color/bg_color_btn1item>
        ...
    </style>

Where Only "@color/bg_color_btn1" will change So how can I reduce this
tedious task??

**Comment by rey5137** [comment]
I have released new 1.1.0 version.  Now it will not ignore `android:background` as before. Please go to [Button Page](https://github.com/rey5137/material/wiki/Button) for more details.
